plugins/outputs/iotdb/iotdb_test.go

```diff
@@ -0,0 +1,305 @@
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const testPort = "6667"
+
+func newTestClient() *IoTDB {
+	testClient := newIoTDB()
+	testClient.Host = "localhost"
+	testClient.Port = testPort
+	testClient.User = "root"
+	testClient.Password = "root"
+	testClient.Log = testutil.Logger{}
+	return testClient
+}
+
+// This func makes sure fields in order. "metric.New" uses map[string]interface{} so fields are not in order.
+func generateTestMetric(
+	name string,
+	tags []telegraf.Tag,
+	fields []telegraf.Field,
+	timestamp time.Time,
+) telegraf.Metric {
+	m := metric.New(name, map[string]string{}, map[string]interface{}{}, timestamp)
+	for _, tag := range tags {
+		m.AddTag(tag.Key, tag.Value)
+	}
+	for _, field := range fields {
+		m.AddField(field.Key, field.Value)
+	}
+	return m
+}
+
+var (
+	constTestTimestamp = time.Date(2022, time.July, 20, 12, 25, 33, 44, time.UTC)
+	testMetrics001     = []telegraf.Metric{
+		generateTestMetric(
+			"root.computer.fan",
+			[]telegraf.Tag{
+				{Key: "price", Value: "expensive"},
+				{Key: "owner", Value: "cpu"},
+			},
+			[]telegraf.Field{
+				{Key: "temperature", Value: float64(42.55)},
+				{Key: "counter", Value: int64(987654321)},
+			},
+			constTestTimestamp,
+		),
+		generateTestMetric(
+			"root.computer.fan",
+			[]telegraf.Tag{ // same keys in different order
+				{Key: "owner", Value: "gpu"},
+				{Key: "price", Value: "cheap"},
+			},
+			[]telegraf.Field{
+				{Key: "temperature", Value: float64(56.24)},
+				{Key: "counter", Value: int64(123456789)},
+			},
+			constTestTimestamp,
+		),
+		generateTestMetric(
+			"root.computer.keyboard",
+			[]telegraf.Tag{},
+			[]telegraf.Field{
+				{Key: "temperature", Value: float64(30.33)},
+				{Key: "counter", Value: int64(123456789)},
+				{Key: "unsigned_big", Value: uint64(math.MaxInt64 + 1000)},
+				{Key: "string", Value: "Made in China."},
+				{Key: "bool", Value: bool(false)},
+				{Key: "int_text", Value: "123456789011"},
+			},
+			constTestTimestamp,
+		),
+	}
+	testMetrics002 = []telegraf.Metric{
+		generateTestMetric(
+			"root.computer.mouse",
+			[]telegraf.Tag{},
+			[]telegraf.Field{
+				{Key: "unsigned_big", Value: uint64(math.MaxInt64 + 1000)},
+			},
+			constTestTimestamp,
+		),
+	}
+)
+
+// compare two recordsWithTags, returns True if and only if they are the same.
+func requireRecordsEqual(t *testing.T, rwtExcepted *recordsWithTags, rwtActual *recordsWithTags, msg string) {
+	t.Logf("Testing case named %q", msg)
+	require.EqualValues(t, rwtExcepted.DeviceIDList, rwtActual.DeviceIDList)
+	require.EqualValues(t, rwtExcepted.MeasurementsList, rwtActual.MeasurementsList)
+	require.EqualValues(t, rwtExcepted.ValuesList, rwtActual.ValuesList)
+	require.EqualValues(t, rwtExcepted.DataTypesList, rwtActual.DataTypesList)
+	require.EqualValues(t, rwtExcepted.TimestampList, rwtActual.TimestampList)
+}
```

Comment:
Please do this in every test instead of using this wrapper. It only makes searching the cause more difficult.

Reply:
All right. This is a reasonable and wise consideration. Thanks a lot for your suggestions.